src/main/java/org/cyclonedx/util/deserializer/VulnerabilityDeserializer.java

```diff
@@ -18,34 +18,299 @@
+        List<Tool> tools = toolsDeserializer.deserialize(toolsParser, ctxt);
+        vulnerability.setTools(tools);
+      }
+    }
+
+    return vulnerability;
+  }
+
+  private void parseAnalysis(JsonNode analysisNode, Vulnerability vulnerability, ObjectMapper mapper) {
+    if (analysisNode != null) {
+      Vulnerability.Analysis analysis = new Vulnerability.Analysis();
+      if (analysisNode.has("state")) {
+        analysis.setState(mapper.convertValue(analysisNode.get("state"), Vulnerability.Analysis.State.class));
+      }
+      if (analysisNode.has("justification")) {
+        analysis.setJustification(mapper.convertValue(analysisNode.get("justification"), Vulnerability.Analysis.Justification.class));
+      }
+      if (analysisNode.has("detail")) {
+        analysis.setDetail(analysisNode.get("detail").asText());
+      }
+      if (analysisNode.has("firstIssued")) {
+        analysis.setFirstIssued(TimestampUtils.parseTimestamp(analysisNode.get("firstIssued").textValue()));
+      }
+      if (analysisNode.has("lastUpdated")) {
+        analysis.setLastUpdated(TimestampUtils.parseTimestamp(analysisNode.get("lastUpdated").textValue()));
+      }
+      if (analysisNode.has("response")) {
+        JsonNode responsesNode = analysisNode.get("response");
+        if (responsesNode.isArray()) {
+          List<Vulnerability.Analysis.Response> responses = mapper.convertValue(responsesNode,
+                  new TypeReference<List<Vulnerability.Analysis.Response>>() {});
+          analysis.setResponses(responses);
+        }
+      } else if (analysisNode.has("responses")) {
+        JsonNode responsesNode = analysisNode.get("responses");
+        if (responsesNode.has("response")) {
+          JsonNode responseNode = responsesNode.get("response");
+          if (responseNode.isArray()) {
+            List<Vulnerability.Analysis.Response> responses = mapper.convertValue(responseNode,
+                    new TypeReference<List<Vulnerability.Analysis.Response>>() {});
+            analysis.setResponses(responses);
+          }
+          else if (responseNode.isTextual()) {
+            Vulnerability.Analysis.Response response = Vulnerability.Analysis.Response.fromString(responseNode.asText());
+            analysis.setResponses(Collections.singletonList(response));
+          }
+        }
+      }
+      vulnerability.setAnalysis(analysis);
+    }
+  }
+
+  private void parseCredits(JsonNode creditsNode, Vulnerability vulnerability, ObjectMapper mapper) {
+    Vulnerability.Credits credits = new Vulnerability.Credits();
+    if (creditsNode.has("organizations")) {
+      parseOrganizations(creditsNode.get("organizations"), credits, mapper);
+    }
+    if (creditsNode.has("individuals")) {
+      parseIndividuals(creditsNode.get("individuals"), credits, mapper);
+    }
+    vulnerability.setCredits(credits);
+  }
+
+  private void parseOrganizations(JsonNode organizationsNode, Vulnerability.Credits credits, ObjectMapper mapper) {
```

Comment:
I think we have something similar in other deserializer

Reply:
> I think we have something similar in other deserializer

Where? 😅. I've looked but the only thing I could find was this [code](https://github.com/shaikhu/cyclonedx-core-java/blob/master/src/main/java/org/cyclonedx/util/deserializer/SignatoryDeserializer.java#L69..L78), which converts a single Organization, where the code here is converting a list of organizations so is a little more complex.

Reply:
you are right, this was removed with some changes